src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -724,6 +807,279 @@ public final void markDirty()
+        return needsWeapon;
+    }
+
+    /**
+     * Check the required pickaxe level..
+     * @return the mining level of the pickaxe.
+     */
+    public int getNeededPickaxeLevel()
+    {
+        return needsPickaxeLevel;
+    }
+
+    /**
+     * Set if the worker needs a shovel.
+     * @param needsShovel true or false.
+     */
+    public void setNeedsShovel(final boolean needsShovel)
+    {
+        this.needsShovel = needsShovel;
+    }
+
+    /**
+     * Set if the worker needs a axe.
+     * @param needsShovel true or false.
+     */
+    public void setNeedsAxe(final boolean needsAxe)
+    {
+        this.needsAxe = needsAxe;
+    }
+
+    /**
+     * Set if the worker needs a hoe.
+     * @param needsShovel true or false.
+     */
+    public void setNeedsHoe(final boolean needsHoe)
+    {
+        this.needsHoe = needsHoe;
+    }
+
+    /**
+     * Set if the worker needs a pickaxe.
+     * @param needsShovel true or false.
+     */
+    public void setNeedsPickaxe(final boolean needsPickaxe)
+    {
+        this.needsPickaxe = needsPickaxe;
+    }
+
+    /**
+     * Set if the worker needs a weapon.
+     * @param needsShovel true or false.
+     */
+    public void setNeedsWeapon(final boolean needsWeapon)
+    {
+        this.needsWeapon = needsWeapon;
+    }
+
+    /**
+     * Add a neededItem to the currentlyNeededItem list.
+     * @param stack the stack to add.
+     */
+    public void addNeededItems(@Nullable ItemStack stack)
+    {
+        if(stack != null)
+        {
+            itemsCurrentlyNeeded.add(stack);
+        }
+    }
+
+    /**
+     * Getter for the neededItems.
+     * @return an unmodifiable list.
+     */
+    public List<ItemStack> getNeededItems()
+    {
+        return Collections.unmodifiableList(itemsCurrentlyNeeded);
+    }
+
+    /**
+     * Getter for the first of the currentlyNeededItems.
+     * @return copy of the itemStack.
+     */
+    @Nullable
+    public ItemStack getFirstNeededItem()
+    {
+        if(itemsCurrentlyNeeded.isEmpty())
+        {
+            return null;
+        }
+        return itemsCurrentlyNeeded.get(0).copy();
+    }
+
+    /**
+     * Clear the currentlyNeededItem list.
+     */
+    public void clearNeededItems()
+    {
+        itemsCurrentlyNeeded.clear();
+    }
+
+    /**
+     * Overwrite the itemsCurrentlyNeededList with a new one.
+     * @param newList the new list to set.
+     */
+    public void setItemsCurrentlyNeeded(@NotNull List<ItemStack> newList)
+    {
+        this.itemsCurrentlyNeeded = new ArrayList<>(newList);
+    }
+
+    /**
+     * Set the needed pickaxe level of the worker.
+     * @param needsShovel the mining level.
+     */
+    public void setNeedsPickaxeLevel(final int needsPickaxeLevel)
+    {
+        this.needsPickaxeLevel = needsPickaxeLevel;
+    }
+
+    /**
+     * Check for the required tool and return the describing string.
+     * @return the string of the required tool.
+     */
+    public String getRequiredTool()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getRequiredTool" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)